Src/Chill/GivenWhenThen.cs

```diff
@@ -183,30 +184,30 @@ protected void WhenLater(Func<Task> whenActionAsync)
         /// <param name="deferredExecution">Should the test be executed immediately or be deferred?</param>
         public void When(Func<Task> whenActionAsync, bool? deferredExecution = null)
         {
-            this.When(() => whenActionAsync.ExecuteInDefaultSynchronizationContext(), deferredExecution);
+            When(() => whenActionAsync.ExecuteInDefaultSynchronizationContext(), deferredExecution);
```

Comment:
We already call ExeuteInDefaultSync... inside TriggerTest in the base class. This can be replaced with only GetAwaiter().GetResult()